modules/redistable/redis_table.c

```
@@ -0,0 +1,938 @@
+        RedisModuleCallReply *e = RedisModule_CallReplyArrayElement(r, i);
+        RedisModuleString *id = RedisModule_CreateStringFromCallReply(e);
+        RedisModule_DictSet(dict, id, NULL);
+    }
+}
+
+// Filter dictionary based on comparison operator
+static void dict_filter_condition(RedisModuleCtx *ctx, RedisModuleDict *dict, RedisModuleString *table,
+                                  RedisModuleString *col, const char *op, RedisModuleString *val) {
+    RedisModuleString *toRemove[1000];
+    size_t removeCount = 0;
+    
+    // Get column type
+    RedisModuleKey *schemaKey = RedisModule_OpenKey(ctx, fmt(ctx, "schema:%s", table), REDISMODULE_READ);
+    RedisModuleString *typeStr = NULL;
+    RedisModule_HashGet(schemaKey, REDISMODULE_HASH_NONE, col, &typeStr, NULL);
+    const char *type = "string";
+    if (typeStr) {
+        size_t tlen; const char *t = RedisModule_StringPtrLen(typeStr, &tlen);
+        if (tlen == 7 && strncasecmp(t, "integer", 7) == 0) type = "integer";
+        else if (tlen == 5 && strncasecmp(t, "float", 5) == 0) type = "float";
+        else if (tlen == 4 && strncasecmp(t, "date", 4) == 0) type = "date";
+    }
+    
+    size_t vlen; const char *vstr = RedisModule_StringPtrLen(val, &vlen);
+    
+    RedisModuleDictIter *it = RedisModule_DictIteratorStartC(dict, "^", NULL, 0);
+    RedisModuleString *key; void *dummy;
+    while ((key = RedisModule_DictNext(ctx, it, &dummy)) != NULL) {
+        RedisModuleString *rowKey = fmt2(ctx, "%s:%s", table, key);
+        RedisModuleCallReply *v = RedisModule_Call(ctx, "HGET", "ss", rowKey, col);
+        int keep = 0;
+        if (v && RedisModule_CallReplyType(v) == REDISMODULE_REPLY_STRING) {
+            RedisModuleString *cur = RedisModule_CreateStringFromCallReply(v);
+            size_t clen; const char *cstr = RedisModule_StringPtrLen(cur, &clen);
+            keep = compare_values(cstr, vstr, op, type);
+        }
+        if (!keep && removeCount < 1000) {
```

This is a **SILENT FAILURE**.

**Risk:** If >1000 rows need filtering, excess rows are silently retained. This violates query correctness.

---

Fixed on next push